### PR TITLE
Fulton Recovery System

### PR DESCRIPTION
**Pull request by Iamgoofball**
https://www.dropbox.com/s/efmk21ri0qtlexq/watch_out_boss_thats_an_enemy_box.mp4?dl=0
https://www.dropbox.com/s/i4pjj1nd1vda0z7/extracted_chemistry_x1.mp4?dl=0
Demo Videos

Since we're in a freeze, I'm doing what incoming did with the parrot talking PR thing and put this up for discussion and code review.

This PR adds the Fulton Recovery System to the game.

https://en.wikipedia.org/wiki/Fulton_surface-to-air_recovery_system

In real life, the gist of it is a balloon is strapped to something, and a plane/helicopter/whatever catches it, grabs the cargo, and drops it off somewhere.

In spacemen, you strap a balloon to something, it flies off into the space sky, and gets dropped off at the point you specify with a beacon for fast delivery of pretty much anything not bolted to the ground(literally).

You get a pack and a beacon, place the beacon somewhere, click on something with the pack after linking it to the beacon, and it'll transport what you clicked on to the beacon via the balloon after a short delay. The videos show it better than I can describe it, I'd recommend watching them.

How it works:
- Fulton Recovery Beacon: A beacon that can be built by R&D to serve as a drop point for Fulton Recovery attempts. Objects fultonned to the beacon will be dropped in a random place in a 3x3 square centered on the beacon.
  - The Syndicate have a beacon version that can only be accessed by Syndicate Fulton Recovery Packs.
- Fulton Recovery Pack: A pack for using the Fulton Recovery System, buildable in R&D. Use it in-hand to link it to a beacon, then click on someone or something that isn't anchored down. After 5 seconds a charge will be used up, and a Fulton Balloon will be strapped to the atom you used it on, and it will be delivered to the beacon you linked it to shortly.
  - R&D can produce an Advanced Fulton Recovery Pack with 24 uses.
  - High tier R&D can also make Super Fulton Recovery Packs with a grand total of **48 uses**.
  - There's also a syndicate version that can fulton targets to the previously mentioned syndicate fulton beacons, and has an insane amount of charges.

I plan to give Traitors access to Fulton Recovery Kits with the syndicate variations.

Let me know what you plan to do with fultons, be it kidnappings, murders, quick prisoner transport, delivering ores back to cargo in record speed, or more!

### CODE REVIEW

**Comment by MrStonedOne** [comment]
Maybe you should start your pr by explain what the fuck a "Fulton Recovery System" is

You know, for those of us who don't know.

**Comment by Iamgoofball** [comment]
i knew I forgot to copypaste in something, let me get the shit I wrote up for that out

**Comment by WJohn** [comment]
In MGS V, fultons are these little balloon strap-ons that you attach to anyone and anything to have it hauled away to your base at like a million miles per hour. This can pick anything up from puppies to people to trucks.

**Comment by MrStonedOne** [comment]
@WJohn I didn't ask for an explanation, I asked for goof to make a better pr description. I know your trying to help, but it doesn't change the fact that that should be in the op, not in the comments.

**Comment by PKPenguin321** [comment]
i remember the first preview of this 10,000 years ago
good times

**Comment by Iamgoofball** [comment]
I put a description in the OP explaining what it is and how it works

File: code/modules/fulton/fulton.dm [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+/obj/item/weapon/extraction_pack
+	name = "Fulton Recovery Pack"
+	desc = "A balloon that can be used to extract a target to a Fulton Recovery Beacon. Anything not bolted down can be moved. Link the pack to a beacon by using the pack in hand."
+	icon = 'icons/obj/fulton.dmi'
+	icon_state = "extraction_pack"
+	var/obj/machinery/extraction_point/beacon
+	var/is_syndie = 0
+	var/uses_left = 12
+
+/obj/item/weapon/extraction_pack/examine()
```
Comment (RemieRichards) [inline code/modules/fulton/fulton.dm]: ```examine(mob/user)```
dont use ```usr```.

File: code/modules/fulton/fulton.dm [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	else
+		var/A
+		if(is_syndie)
+			A = input("Select a beacon to connect to", "Balloon Extraction Pack", A) in total_extraction_beacons
+		else
+			A = input("Select a beacon to connect to", "Balloon Extraction Pack", A) in normal_extraction_beacons
+		if(!A)
+			return
+		beacon = A
+		return
```
Comment (RemieRichards) [inline code/modules/fulton/fulton.dm]: return at the end is needless here, you know this.

File: code/modules/fulton/fulton.dm [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+			animate(holder_obj, pixel_z = 1000, time = 30)
+			if(istype(A, /mob/living/carbon/human))
+				var/mob/living/carbon/human/L = A
+				L.forcesay(extraction_appends) // ";HELP BL-AAAAAAAAAAAAAAAAHHHHHHHHH"
+				L.SetParalysis(0) // wakey wakey
+				L.drowsyness = 0
+				L.sleeping = 0
+			sleep(30)
+			var/list/flooring_near_beacon = list()
+			for(var/turf/simulated/floor/floor in orange(1, beacon))
```
Comment (RemieRichards) [inline code/modules/fulton/fulton.dm]: ```trange()``` is better

File: code/modules/fulton/fulton.dm [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+			holder_obj.overlays -= balloon
+			A.anchored = 0 // An item has to be unanchored to be extracted in the first place.
+			A.density = initial(A.density)
+			animate(holder_obj, pixel_z = 0, time = 5)
+			sleep(5)
+			A.loc = holder_obj.loc
+			qdel(holder_obj)
+			if(uses_left <= 0)
+				qdel(src)
+			return
```
Comment (RemieRichards) [inline code/modules/fulton/fulton.dm]: useless return

File: code/modules/fulton/fulton.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: it satisfies my OCD so it stays

File: code/modules/fulton/fulton.dm (hunk 4 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: satisfies my OCD, it stays

File: code/modules/fulton/fulton.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: will do

File: code/modules/fulton/fulton.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: good to know we support that now, i hated using usr but I thought it was the only way still for examine

File: code/modules/fulton/fulton.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/fulton/fulton.dm]: No goof, not happening.

**Comment by Metacide** [comment]
Delightful :+1:

File: code/modules/fulton/fulton.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: o c d

Comment (RemieRichards) [inline code/modules/fulton/fulton.dm]: P R CLOSED.

**Comment by GunHog** [comment]
Stealing all of the armory made easy!

File: code/modules/fulton/fulton.dm [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	uses_left = 48
+
+/obj/item/weapon/extraction_pack/syndicate
+	name = "syndicate fulton recovery pack"
+	is_syndie = 1
+	uses_left = 9999
+
+/obj/item/weapon/extraction_pack/attack_self(mob/user)
+	if(!total_extraction_beacons.len)
+		user << "There are no extraction beacons in existance!"
```
Comment (bgobandit) [inline code/modules/fulton/fulton.dm]: existence

**Comment by Fox-McCloud** [comment]
This is incredibly and insanely powerful; you can space lotssss of people with a single one of these and a stunprod.

it should probably have a check that disallow it to work if the beacon is in a hostile environment (similar to the holoparasite's teleport mechanic).

Will also definitely be used to rapid teleport all over the place.

File: code/modules/fulton/fulton.dm [hunk 6]
```diff
@@ -0,0 +1,152 @@
+var/list/normal_extraction_beacons = list()
+var/list/total_extraction_beacons = list()
+
+/obj/item/weapon/extraction_pack
+	name = "Fulton Recovery Pack"
```
Comment (MrPerson) [inline code/modules/fulton/fulton.dm]: Needs to be lower case

File: code/modules/fulton/fulton.dm [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+				flooring_near_beacon += floor
+			holder_obj.loc = pick(flooring_near_beacon)
+			animate(holder_obj, pixel_z = 10, time = 50)
+			sleep(50)
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			holder_obj.overlays -= balloon
+			A.anchored = 0 // An item has to be unanchored to be extracted in the first place.
```
Comment (MrPerson) [inline code/modules/fulton/fulton.dm]: It may have become anchored after the do_after(), however

**Comment by MrPerson** [comment]
I'm of course going to reiterate my point that these are magic balloons flying through the ceiling of the space station, which is incredibly silly and retarded.

Just make it a simple teleportation dealio. Just because you have a fetish for ripping off MGS doesn't mean fultons are better flavorwise.

File: code/modules/fulton/fulton.dm (hunk 7 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: It is anchored after the do_after finishes so you can't pick it up/mess with it at all while it's being fultonned. You shouldn't be able to interact with or anchor it manually during the fulton process.

**Comment by KazeEspada** [comment]
no, no, there are rails in the ceiling and these claws of life travel along them.

**Comment by Fox-McCloud** [comment]
Obviously bluespace portals are being generated which they shoot through.

File: code/modules/fulton/fulton.dm (hunk 7 above)
Comment (MrPerson) [inline code/modules/fulton/fulton.dm]: No I mean you can unwrench a machine, attach a magic balloon to it, and have someone else wrench it down during the process. Then the balloon carries it off while it's anchored and unanchores it afterwards. Which can lead to all sorts of weird situations.

Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: Good point, I'll add some handling shit for that to end the fulton attempt there.

**Comment by Lumipharon** [comment]
Time to fulton everyone into plasma fire/the tesla engine/plain old space with this + prod.
Also ballons flying shit off on a space station is memetastic and silly.

File: code/modules/fulton/fulton.dm [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+	if(!istype(A))
+		return
+	else
+		if(A.loc == user || A == user) // no extracting stuff you're holding in your hands/yourself
+			return
+		if(A.anchored)
+			return
+		user << "<span class='notice'>You start attaching the pack to [A]...</span>"
+		if(do_after(user,50,target=A))
+			user << "<span class='notice'>You attach the pack to [A] and activate it.</span>"
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Check if A still exists.

File: code/modules/fulton/fulton.dm [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+			var/obj/effect/extraction_holder/holder_obj = new(A.loc)
+			holder_obj.appearance = A.appearance
+			A.loc = holder_obj
+			balloon = image(icon,"extraction_balloon")
+			balloon.pixel_y = 10
+			balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			holder_obj.overlays += balloon
+			playsound(holder_obj.loc, 'sound/items/fulext_deploy.wav', 50, 1, -3)
+			animate(holder_obj, pixel_z = 10, time = 20)
+			sleep(20)
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Check if A still exists.

File: code/modules/fulton/fulton.dm [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Check if A still exists

File: code/modules/fulton/fulton.dm [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Check if...

File: code/modules/fulton/fulton.dm [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Every sleep, you have to check if A still exists.

File: code/modules/fulton/fulton.dm [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			playsound(holder_obj.loc, 'sound/items/fultext_launch.wav', 50, 1, -3)
+			animate(holder_obj, pixel_z = 1000, time = 30)
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Another check for A. To make sure it still exists.

File: code/modules/fulton/fulton.dm (hunk 7 above)
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Another check right before here to make sure it still exists.

**Comment by phil235** [comment]
This is retarded, make it a teleportation device like MrPerson said.

File: code/_globalvars/lists/flavor_misc.dm [hunk 14]
```diff
@@ -35,6 +35,7 @@ var/global/list/female_clothing_icons = list()
 
 	//radical shit
 var/list/hit_appends = list("-OOF", "-ACK", "-UGH", "-HRNK", "-HURGH", "-GLORF")
+var/list/extraction_appends = list("-AAAAAAAAAAAAAAAAAUGH", "-AAAAAAAAAAAHHHHHHHHHH")
```
Comment (AnturK) [inline code/_globalvars/lists/flavor_misc.dm]: This really shouldn't be global.

**Comment by xxalpha** [comment]
Remember the station has a ceiling.

:-1:

**Comment by Iamgoofball** [comment]
@phil235 When I get the sprites for it, I'll have a wormhole thing situated above the balloon and one above the output spot so it goes through that.

I 100% refuse to make it "yet another "make spark system then change loc" teleport". That shit is boring. The whole point of this was to have a fun visual effect. I'll do the compromise with the wormholes but that's as much as you're getting.

@xxalpha what ceiling, I don't see one on my screen

File: code/_globalvars/lists/flavor_misc.dm (hunk 14 above)
Comment (Iamgoofball) [inline code/_globalvars/lists/flavor_misc.dm]: figured I'd put it with the other appends because it used the same system and shit

**Comment by AlexCrimson** [comment]
This is just ripping features from a game you like. Doesnt fit with SS13 at all.

**Comment by Fox-McCloud** [comment]
>This is just ripping features from a game you like. Doesnt fit with SS13 at all.

Thematically, maybe, but functionally, it seems perfectly valid to me---and Goof does have a strong point of "lulz, sparks and disappear to new location" being...overused.

File: code/_globalvars/lists/flavor_misc.dm (hunk 14 above)
Comment (AnturK) [inline code/_globalvars/lists/flavor_misc.dm]: The other appends are used in more than one item.

**Comment by duncathan** [comment]
I personally love this
It's silly and that's great because this whole game sure as hell is

**Comment by PKPenguin321** [comment]
Yeah, gameplay > realism wins here

**Comment by xxalpha** [comment]
The item's function does not have to be altered at all by simply changing how it looks. How it looks is important.

**Comment by bgobandit** [comment]
What is the functional difference between this and telescience?

**Comment by xxalpha** [comment]
Well you need to physically be near the thing you want to teleport.

It will still be pretty fucking overpowered for shit like taking people to the brig or teleporting someone to a death trap.

**Comment by PKPenguin321** [comment]
Personally I can't wait to place the beacon in front of an emagged crusher

**Comment by GunHog** [comment]
>What is the functional difference between this and telescience?

People do not need outside tools to operate the Fulton system.

File: code/modules/fulton/fulton.dm (hunk 8 above)
Comment (MrStonedOne) [inline code/modules/fulton/fulton.dm]: No need, do_after returns false if a disappears

**Comment by AlexanderUlanH** [comment]
>SS13 having no ceiling is now canon

**Comment by PKPenguin321** [comment]
@AlexanderUlanH 
>When I get the sprites for it, I'll have a wormhole thing situated above the balloon and one above the output spot so it goes through that.

File: code/modules/fulton/fulton.dm (hunk 8 above)
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: Ah, okay, so not here, but there are other sleeps later.

**Comment by Iamgoofball** [comment]
keep in mind you can't fulton yourself

**Comment by Dorsisdwarf** [comment]
:-1:

1) Fulton is such a stupid name. Give it a more generic name so you dont have to hope the players MSGV or read the wiki article

2) So many fucking uses. 12 by default, 48 by R&D work? naaaaaah

3) R&D dont even need this. Its literally useless for anything except maybe medical doctors, who wont ever get it because R&D will be stealing things with it instead.

4) What's the use of this for syndies? Having an instant-valid item to dump your objectives somewhere assistant mcgee can steal them?

**Comment by GunHog** [comment]
> 1) Fulton is such a stupid name. Give it a more generic name so you dont have to hope the players played MSGV or read the wiki article

Meh, this is a minor nitpick at best.

> 2) So many fucking uses. 12 by default, 48 by R&D work? naaaaaah

This may be a valid point, you could clear out entire rooms with these things!

> 3) R&D dont even need this. Its literally useless for anything except maybe medical doctors, who wont ever get it because R&D will be stealing things with it instead.

Give one to Sec and CMO as default, R&D can make more for them. Scientists would do well to hand some out to cargo as well, for getting mineral deliveries faster!

> 4) What's the use of this for syndies? Having an instant-valid item to dump your objectives somewhere assistant mcgee can steal them?

Blast in the armory, Steal everything you can carry, send away the rest of the loot. Hide bodies the same way. If you send stuff to an insecure location, it is your fault if it gets discovered.

**Comment by duncathan** [comment]
Fulton is a generic name when it's literally a real life thing

**Comment by Iamgoofball** [comment]
Fultons are literally a thing the CIA tried to use.

Also
>Its literally useless for anything except maybe medical doctors,

AHAHA no, there's shittons of uses

**Comment by Dorsisdwarf** [comment]
What I mean is something like "portable airlift system", @duncathan 

I'm aware they're a real thing goofball. They worked, too, and were surprisingly unfatal.

And when I say "literally useless" I mean "No legitimate use". What are you gonna use an instant-steal for as a scientist except for being a shitter? Sec can have it too, maybe.

Except that sec having an item that lets them transport an entire room of people to a location almost instantly without having to drag them through the halls is a bad idea. Remember the fuss about the DRAGnet? This is the DRAGnet except you get ten times as many uses.

**Comment by TrustyGun** [comment]
There's ton of legit uses for this, Dorsidwarf:

* Miners transporting minerals to science
* Doctors saving people much more effectively
* Delivering objects is so much more easier
* Delivering objects to someone in different z-level (like an away mission)

Also, not every researchable item is supposed to be used for just the scientists. Take a look at the Janitorial Cart Upgrade, all the mining shit, and even all the security buffs science can make, which obviously can't be used by them. Assuming that every item in RnD should be a benefit to just science is stupid.

**Comment by GunHog** [comment]
Indeed, such as Kor's thing that will add slime speed pot functionality to vehicles. That is useful to Sec and the Janitor - Science does not have vehicles, and cannot make them (barring mechs, but they are not supported by the potion).

**Comment by xxalpha** [comment]
@TrustyGun I would use those arguments against this.

Reminder that this is a game about people interacting with other people.

**Comment by Iamgoofball** [comment]
why don't we try it out for awhile and see how it plays out instead of theorycrafting everything till it's ruined

**Comment by Dorsisdwarf** [comment]
Your counterarguments are, in order 
1) Making the least social job on the station even less social (and exposed)
2) Something I already specifically mentioned
3)You can't deliver until you've been there, so it's only good for taking things /back/ to your science fortress, unless science need to mass-deliver something for some reason
4) this is not a good thing.

File: code/modules/research/designs/weapon_designs.dm [hunk 15]
```diff
@@ -272,3 +272,33 @@
...
+	category = list("Weapons")
+
+/datum/design/extraction_pack_super
+	name = "Super Fulton Recovery Pack"
+	desc = "A fulton recovery pack with 48 uses."
+	id = "extraction_pack_super"
+	req_tech = list("engineering" = 4)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 4000, MAT_GLASS = 5000)
+	build_path = /obj/item/weapon/extraction_pack/super
```
Comment (Dorsisdwarf) [inline code/modules/research/designs/weapon_designs.dm]: 48 grabs of absolutely anything, at the low, low price of TWO sheets of glass and metal each.
So super, the only req tech for the highest tier is engi 4.

File: code/modules/fulton/fulton.dm [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+	uses_left = 24
+
+/obj/item/weapon/extraction_pack/super
+	name = "super fulton recovery pack"
+	uses_left = 48
+
+/obj/item/weapon/extraction_pack/syndicate
+	name = "syndicate fulton recovery pack"
+	is_syndie = 1
+	uses_left = 9999
```
Comment (Dorsisdwarf) [inline code/modules/fulton/fulton.dm]: is this a joke.

**Comment by Dorsisdwarf** [comment]
I'm not against the idea of the fultons in itself, I'm mostly against the ridiculous spammage and the fact that R&D, the Department of Keeping Cool Stuff To Yourself, gets them when they need them least.

File: code/modules/fulton/fulton.dm (hunk 16 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: No, this is for the traitor item edition which shouldn't run out of charges.

**Comment by tkdrg** [comment]
I hate the idea. Magic teleportation is boring and non-interactive for traitors, it basically removes a lot of risk and doesn't add any good gameplay. It feels like the only point of this PR is MGS memes

**Comment by AlexCrimson** [comment]
Its Goofball. When does he not add stupid meme shit to everything he codes or says? Le kewl internet culture right?

**Comment by Niknakflak** [comment]
Preach it

**Comment by Iamgoofball** [comment]
We ran a test run of this as an event, and players loved it, going by poll results for people who actually bothered to vote.

I'm going to have a few more rounds ran with this being tested.

**Comment by YotaXP** [comment]
Don't fulton balloons require an aircraft to intercept and grab it from the sky?

File: code/modules/fulton/fulton.dm (hunk 16 above)
Comment (Jordie0608) [inline code/modules/fulton/fulton.dm]: This is not how you make something infinite use and you know it

**Comment by duncathan** [comment]
Who needs an aircraft when we have bluespace

**Comment by KorPhaeron** [comment]
@tkdrg 

Can we make them admin only so I can give them to lavaland miners to fulton back ore crates and other gear they discover in the ruins (which I will totally map soon I swear)

File: code/modules/fulton/fulton.dm [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+		var/A
+		if(is_syndie)
+			A = input("Select a beacon to connect to", "Fulton Recovery Pack", A) in total_extraction_beacons
+		else
+			A = input("Select a beacon to connect to", "Fulton Recovery Pack", A) in normal_extraction_beacons
+		if(!A)
+			return
+		beacon = A
+
+/obj/item/weapon/extraction_pack/afterattack(atom/movable/A, mob/living/carbon/human/user, flag, params)
```
Comment (KazeEspada) [inline code/modules/fulton/fulton.dm]: This should be attack() not afterattack(), you generally want to stop the normal attack_by() if successful.

**Comment by KazeEspada** [comment]
Fultons use the layer of the object they grab, which for items is lower than mobs and looks like shit.

**Comment by Shadowlight213** [comment]
A few issues.
You can build a fulton point on the supply shuttle and fulton to centcom
Fultonning an object someone if buckled to doesn't fulton them and leaves them buckled to an invisible object.
If you unset anchored on a fulton point via VV and then fulton the fulton point with the target set to the fulton point you are fultonning, the point will rise up into the sky and disappear. It will then not show up on the target select screen.
However, if you don't change targets and fulton someone, they will have the animation occur, but wont be teleported and will be forever stuck invisible at the original location.

**Comment by KazeEspada** [comment]
@Shadowlight213 VV doesn't count.

**Comment by Shadowlight213** [comment]
K

**Comment by KorPhaeron** [comment]
@Iamgoofball 

Just make it admin only, dunc said the code was fine.

Fix the issues with putting them on the cargo shuttle first though please.

**Comment by duncathan** [comment]
Mostly fine
Review incoming
I want this merged goddammit

File: code/modules/fulton/fulton.dm [hunk 18]
```diff
@@ -0,0 +1,163 @@
+var/list/normal_extraction_beacons = list()
+var/list/total_extraction_beacons = list()
+var/list/extraction_appends = list("-AAAAAAAAAAAAAAAAAUGH", "-AAAAAAAAAAAHHHHHHHHHH")
```
Comment (duncathan) [inline code/modules/fulton/fulton.dm]: All three of these can be `var/global` on the extraction pack

File: code/modules/fulton/fulton.dm [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+					qdel(src)
+				return
+			A.anchored = 0 // An item has to be unanchored to be extracted in the first place.
+			A.density = initial(A.density)
+			animate(holder_obj, pixel_z = 0, time = 5)
+			sleep(5)
+			A.loc = holder_obj.loc
+			qdel(holder_obj)
+			if(uses_left <= 0)
+				qdel(src)
```
Comment (duncathan) [inline code/modules/fulton/fulton.dm]: You copy this maybe four times total. Make it a proc. Override it to just return on the syndie one so you can get rid of that ridiculous 9999.

File: code/modules/fulton/fulton.dm (hunk 18 above)
Comment (duncathan) [inline code/modules/fulton/fulton.dm]: Scratch that, put total on the syndie pack

File: code/modules/fulton/fulton.dm [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+/obj/item/weapon/extraction_pack/attack_self(mob/user)
+	if(!total_extraction_beacons.len)
+		user << "There are no extraction beacons in existence!"
+		return
+	else
+		var/A
+		if(is_syndie)
+			A = input("Select a beacon to connect to", "Fulton Recovery Pack", A) in total_extraction_beacons
+		else
+			A = input("Select a beacon to connect to", "Fulton Recovery Pack", A) in normal_extraction_beacons
```
Comment (duncathan) [inline code/modules/fulton/fulton.dm]: Replace this with a car pointing to which list it picks the beacon from. Kill `is_syndie`.

File: code/modules/fulton/fulton.dm (hunk 17 above)
Comment (duncathan) [inline code/modules/fulton/fulton.dm]: Yes please

File: code/modules/fulton/fulton.dm (hunk 20 above)
Comment (Iamgoofball) [inline code/modules/fulton/fulton.dm]: >a car

when did we add cars

Comment (Shadowlight213) [inline code/modules/fulton/fulton.dm]: Beep beep!
I'm guessing he meant var though

File: code/modules/fulton/fulton.dm [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+				M.buckled = 0 // Unbuckle them to prevent anchoring problems
+			else
+				A.anchored = 1
+				A.density = 0
+			var/obj/effect/extraction_holder/holder_obj = new(A.loc)
+			holder_obj.appearance = A.appearance
+			A.loc = holder_obj
+			balloon = image(icon,"extraction_balloon")
+			balloon.pixel_y = 10
+			balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
```
Comment (duncathan) [inline code/modules/fulton/fulton.dm]: `/image` is a child of `/datum`. Make this balloon stuff a subtype of `/image` and then give extraction packs another `var/global` pointing to the balloon image.

**Comment by Ultimate-Chimera** [comment]
No. This doesn't even make sense for space. :-1:

**Comment by Iamgoofball** [comment]
Doesn't matter.

Ps. Need new sprites before I'm okay with a merge @KorPhaeron

**Comment by KorPhaeron** [comment]
She has 3 commissions queued up now so it'll be a bit before she is ready to DO IT FOR FREE again

**Comment by Iamgoofball** [comment]
oh god

**Comment by Iamgoofball** [comment]
all the spriters refused to give me sprites

**Comment by KorPhaeron** [comment]
My girlfriend just finished her spessmen commissions she said she'd do the balloons now.

**Comment by Iamgoofball** [comment]
YES

**Comment by Iamgoofball** [comment]
# Y
# E
# S

File: code/modules/research/designs/weapon_designs.dm [hunk 22]
```diff
@@ -272,3 +272,33 @@
 	materials = list(MAT_METAL = 2000, MAT_SILVER = 500)
 	build_path = /obj/item/weapon/suppressor
 	category = list("Weapons")
+
+/datum/design/extraction_pack
```
Comment (KorPhaeron) [inline code/modules/research/designs/weapon_designs.dm]: Add the new sprites and remove them from RD so it can finally be merged

**Comment by Cheridan** [comment]
100th comment

**Comment by Dorsisdwarf** [comment]
Wheres muh noo sprites

**Comment by KorPhaeron** [comment]
@Iamgoofball

**Comment by MrStonedOne** [comment]
This has been pending some issues where it causes unprofileable lag that i think has to do with how byond sends out map updates to clients and some edge case this is hitting.

**Comment by Iamgoofball** [comment]
yeah, plus goon code scandal

**Comment by GunHog** [comment]
What does the scandal have to do with your fulton client?

**Comment by Iamgoofball** [comment]
distraction

**Comment by KorPhaeron** [comment]
@Iamgoofball

**Comment by GunHog** [comment]
Why have you not merged this yet?

**Comment by KorPhaeron** [comment]
@GunHog 

It isn't done

**Comment by Iamgoofball** [comment]
yeah I still need to do R&D with @MrStonedOne on the lag issue

**Comment by AnturK** [comment]
No updates for a month. Please reopen when you dealt with the lag and other issues.

**Comment by Thunder12345** [comment]
Nice fucking stale PR bro

**Comment by Shadowlight213** [comment]
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO :cry:

**Comment by GunHog** [comment]
Did you test this without the sleep()s? Could that be the lag source?

**Comment by MrStonedOne** [comment]
its not profiled lag, meaning its lag with send_maps() the byond routine that updates clients, or any of the maintenance work byond does.

most likely something about shifting the pixel_x/y is sending out appearance updates to all clients, not just clients in view, byond's new map chunking system most likely fixed this.

**Comment by Cheridan** [comment]
balloons in space are never not going to be dumb

**Comment by duncathan** [comment]
this is being worked on again

**Comment by Iamgoofball** [comment]
fuck me, conflicts, closing to migrate to new branch since I can't view mergability on a closed pr :^))))))))

**Pull request closed without merging**